### PR TITLE
Release 1.2.9

### PR DESCRIPTION
**Pull request by dylanuys**
## TLDR
- Two new diffusion models for validator challenges
- Better device management, now exposing `--neuron.device` through `validator.env`'s `DEVICE` variable
- Deprecating unused prompt generation model due to the biases it displayed. We have replaced this with our ImageAnnotationGenerator pipeline.

## New Model Details
We are introducing two additional popular open source image models for validator challenges, as part of our continuous effort to meaningfully expand the distribution of synthetic images that miners classify. Our goal is to encourage better miner generalizability to images produced by different deepfake methods.

**For validators:** No new dependencies/install requirements were added - models will automatically be downloaded to your local huggingface cache. 

**For miners:** We foresee a difficulty increase in subnet challenges as miners adapt to the new models' image distributions.

The two models are:

- [prompthero/openjourney-v4](https://huggingface.co/prompthero/openjourney-v4), a popular StableDiffusion model finetuned on +124k Midjourney v4 images, by [PromptHero](https://prompthero.com/?utm_source=huggingface&utm_medium=referral) (158,394 downloads last month as of writing)
- [cagliostrolab/animagine-xl-3.1](https://huggingface.co/cagliostrolab/animagine-xl-3.1), a popular StableDiffusionXL model finetuned on anime images, by [Cagliostro Research Lab](https://huggingface.co/cagliostrolab) in collaboration with [SeaArt.ai](https://www.seaart.ai/) (411,846 downloads last month as of writing)

These were selected by

1. Manually reviewing state-of-the-art models (tech review blogs, AI news, Hugging Face models) and open source models finetuned on proprietary image generators (e.g. Midjourney) and/or popular generated image genres in-the-wild (e.g. anime)
2. Applying our [subnet API benchmarking](https://www.notion.so/bitmindlabs/Subnet-API-Benchmark-13eaf854028380c1bebdcee58508995f) procedure to synthetic mirrors generated by candidate models.
   - Notably, our subnet scored worse on openjourney-v4 and animagine-xl-3.1 (68.76% and 83.8% accuracy) than any of our existing validator challenge models, indicating that miners have yet to optimally generalize to those model output distributions (supporting the addition of these models).
   - We sourced the real images for mirroring through a scraping method outlined in the technical blog post (link above).
   - API benchmarking for new model selection was conducted end-to-end within 24-HR spans to increase the likelihood that accuracy measures reflected a stable subnet state (as opposed to accuracy variations caused by significant miner changes on mainnet).

### CODE REVIEW

**Review by aliang322 — APPROVED** [review]
Looks good! Excited for the new models and clarified flows.
